packages/core-connector-interface/src/channel.d.ts

```diff
@@ -1,216 +1,50 @@
-   * @param props additional arguments
-   */
-  // open(coreConnector: any, amount: Balance.Instance, signature: Promise<Uint8Array>, ...props: any[]): Promise<ChannelInstance>
-
-  /**
-   * Fetches all channel instances from the database and applies first `onData` and
-   * then `onEnd` on the received instances.
-   * @param onData applied on all channel instances
-   * @param onEnd composes at the end the received data
-   */
-  getAll<T, R>(
-    onData: (channel: Channel, ...props: any[]) => Promise<T>,
-    onEnd: (promises: Promise<T>[], ...props: any[]) => R
-  ): Promise<R>
-
-  /**
-   * Get stored channel
-   * @param counterPartyPubKey the public key of the counterparty in which we have a stored channel with
-   * @returns a promise tha resolves into a 'SignedChannel'
-   */
-  getOffChainState(counterparty: Uint8Array): Promise<SignedChannel>
-
-  /**
-   * Get on-chain channel data
-   * @param counterPartyPubKey the public key of the counterparty in which we have a channel with
-   * @returns a promise tha resolves into on chain channel data
-   */
-  getOnChainState(counterparty: Public): Promise<ChannelEntry>
-
-  /**
-   * Fetches all channel instances from the database and initiates a settlement on
-   * each of them.
-   * @param props additional arguments
-   */
-  closeChannels(): Promise<Balance>
-
-  /**
-   * Increases the balance of the payment channel with the given counterparty
-   * by the given amount
-   * @param counterParty the counterparty of the channel
-   * @param amount the amount of tokens to put into the payment channel
-   */
-  increaseFunds(counterParty: Address, amount: Balance): Promise<void>
-
-  /**
-   * Handles a channel opening request.
-   * @notice Takes the `coreConnector` instance and returns an async iterable duplex stream.
-   * @param coreConnector coreConnector instance
-   */
-  handleOpeningRequest(source: AsyncIterable<Uint8Array>): AsyncIterable<Uint8Array>
-
-  /**
-   * Create a signedChannel instance.
-   * @param arr array containing a signedChannel
-   * @param struct desired content of the signedChannel
-   */
-  createSignedChannel(
-    arr?: {
-      bytes: ArrayBuffer
-      offset: number
-    },
-    struct?: {
-      channel: ChannelType
-      signature?: Signature
-    }
-  ): Promise<SignedChannel>
-
-  tickets: {
-    /**
-     * Submits a signed ticket to the blockchain.
-     * @param signedTicket a signed ticket
-     * @param secretA a signed ticket
-     * @param secretB a signed ticket
-     */
-    submit(
-      ticket: AcknowledgedTicket,
-      ticketIndex: Uint8Array
-    ): Promise<
-      | {
-          status: 'SUCCESS'
-          receipt: string
-          ackTicket: AcknowledgedTicket
-        }
-      | {
-          status: 'FAILURE'
-          message: string
-        }
-      | {
-          status: 'ERROR'
-          error: Error | string
-        }
-    >
-  }
+  new (indexer: Indexer, connector: any, self: Public, counterparty: Public): Channel
```

Comment:
keeping it like this for now - will change once https://github.com/hoprnet/hoprnet/pull/1396 is merged